### PR TITLE
CNV-36299: Sort by name NNS

### PR DESCRIPTION
**Pull request by upalatucci**
Sort NNS by name on toggle name column sort icon.

The table is not the `VirtualizedTable` as it does not support the automatic sort. So we need to do it manually. 

patternfly: https://v4-archive.patternfly.org/v4/components/table/#composable-sortable--wrapping-headers

Screenshot from 2023-12-11 11-17-12
Screenshot from 2023-12-11 11-17-08

### CODE REVIEW

File: src/views/states/list/hooks/useSortStates.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  let sortedStates = nodeNetworkStates;
+  if (activeSortIndex === 1) {
+    sortedStates = nodeNetworkStates.sort((a, b) => {
+      const aValue = a.metadata.name;
+      const bValue = b.metadata.name;
+
+      if (activeSortDirection === 'asc') {
+        return (aValue as string).localeCompare(bValue as string);
+      }
+      return (bValue as string).localeCompare(aValue as string);
```
Comment (hstastna) [inline src/views/states/list/hooks/useSortStates.ts]: Just wondering if we couldn't use `as string` a bit earlier to avoid repeating ourselves so much in these lines..
Maybe if we better used
```
const aValue = a.metadata.name as string;
const bValue = b.metadata.name as string;
```
 ?

File: src/views/states/list/StatesList.tsx [hunk 2]
```diff
@@ -95,21 +100,30 @@ const StatesList: FC = () => {
...
 
-          {filteredData.length > 0 ? (
-            <Table
-              cells={activeColumns}
-              rows={paginatedData}
-              gridBreakPoint={TableGridBreakpoint.none}
-              role="presentation"
-            >
-              <TableHeader />
+          {sortedStates.length > 0 ? (
```
Comment (hstastna) [inline src/views/states/list/StatesList.tsx]: Maybe `sortedStates?.length` ? As we use it like that a few lines earlier...

File: src/views/states/list/hooks/useSortStates.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    });
+  }
+
+  const nameSortParams: ThSortType = {
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
```
Comment (hstastna) [inline src/views/states/list/hooks/useSortStates.ts]: Maybe we should add types for `event`, `index` and `direction`?

File: scripts/start-console.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@ set -euo pipefail
 
 npm_package_consolePlugin_name="nmstate-console-plugin"
 
-CONSOLE_IMAGE=${CONSOLE_IMAGE:="quay.io/openshift/origin-console:latest"}
+CONSOLE_IMAGE=${CONSOLE_IMAGE:="quay.io/openshift/origin-console:4.14"}
```
Comment (hstastna) [inline scripts/start-console.sh]: Not sure what is the reason of this change.. Do we really want this?

File: src/views/states/list/hooks/useSortStates.ts (hunk 1 above)
Comment (upalatucci) [reply src/views/states/list/hooks/useSortStates.ts]: Yes sure

File: src/views/states/list/hooks/useSortStates.ts (hunk 3 above)
Comment (upalatucci) [reply src/views/states/list/hooks/useSortStates.ts]: Those inputs are already typed with `ThSortType` on top. If you hover on them, you can see the type

File: src/views/states/list/StatesList.tsx (hunk 2 above)
Comment (upalatucci) [reply src/views/states/list/StatesList.tsx]: Okay

File: scripts/start-console.sh (hunk 4 above)
Comment (upalatucci) [reply scripts/start-console.sh]: This is because on `:latest` we have to refresh everytime the browser data otherwise the plugin will not show up. Just like happens with kubevirt-plugin. In this case our e2e tests fails for this reason. So we can keep using 4.14 and when this issue is resolved, reverse this change.

File: src/views/states/list/hooks/useSortStates.ts (hunk 3 above)
Comment (metalice) [reply src/views/states/list/hooks/useSortStates.ts]: Im with hilda here, i like also to see the types directly instead of searching them on component docs, or hovering :)

**Review by hstastna — APPROVED** [review]
Looks good, all the comments resolved. Approving! Good work!
